stacks-signer/src/client/stackerdb.rs

```diff
@@ -90,8 +115,103 @@ impl StackerDB {
         }
     }
 
+    /// Get the latest signer transactions from signer ids
+    pub fn get_signer_transactions_with_retry(
+        &mut self,
+        signer_ids: &[u32],
+    ) -> Result<Vec<(u32, Vec<StacksTransaction>)>, ClientError> {
+        let slot_ids: Vec<_> = signer_ids
+            .iter()
+            .map(|id| id * SIGNER_SLOTS_PER_USER + TRANSACTIONS_SLOT_ID)
+            .collect();
+        debug!(
+            "Getting latest chunks from stackerdb for the following signers: {:?}",
+            signer_ids
+        );
+        let send_request = || {
+            self.signers_stackerdb_session
+                .get_latest_chunks(&slot_ids)
+                .map_err(backoff::Error::transient)
+        };
+        let chunk_ack = retry_with_exponential_backoff(send_request)?;
+        let mut transactions = Vec::new();
+        for (i, chunk) in chunk_ack.iter().enumerate() {
+            if let Some(data) = chunk {
+                if let Ok(message) = bincode::deserialize::<SignerMessage>(data) {
+                    if let SignerMessage::Transactions(chunk_transactions) = message {
+                        let signer_id = *signer_ids.get(i).expect(
```

Comment:
What happens if `chunk_ack.len() >= signer_ids.len()`?

Reply:
This cannot happen. The call get_latest_chunks will just return None for slots that don't exist so we will always return as many chunks as we queried. and we query an equal number of slots to signer_ids.If this was to happen, it would be a BUG in the system hence the expect.